TheKeyOAuth2/TheKeyOAuth2Client+PasswordGrant.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  TheKeyOAuth2Client+PasswordGrant.swift
+//  TheKeyOAuth2
+//
+//  Created by Ryan Carlson on 12/27/17.
+//  Copyright © 2017 TheKey. All rights reserved.
+//
+
+import Foundation
+import GTMOAuth2
+
+public enum TheKeyPasswordGrantResult {
+    case success, badPassword, jsonParsingError, serverError, clientNotConfiguredError, responseCastingError, unknownError
+}
+
+public extension TheKeyOAuth2Client {
+    public func passwordGrantLogin(for username: String, password: String, completion: @escaping (TheKeyPasswordGrantResult, Error?) -> Void) {
+        if !isConfigured() {
+            completion(.clientNotConfiguredError, nil)
+            return
+        }
+        
+        guard let request = buildAccessTokenRequest(for: username, password: password) else {
+            completion(.clientNotConfiguredError, nil)
+            return
+        }
+        
+        let session = URLSession(configuration: .default, delegate: self, delegateQueue: .main)
+        
+        session.dataTask(with: request) { (data, response, error) in
+            if let error = error {
+                completion(.unknownError, error)
+                return
+            }
+            
+            guard let httpResponse = response as? HTTPURLResponse else {
+                completion(.responseCastingError, nil)
+                return
+            }
+
+            if httpResponse.statusCode == 200, let usableData = data {
+                self.handleSuccessfulPasswordGrant(responseData: usableData, completion: completion)
+                return
+            }
+            
+            if httpResponse.statusCode == 400, let usableData = data {
+                self.handleBadResponse(responseData: usableData, completion: completion)
+                return
+            }
+            
+            if httpResponse.statusCode / 100 == 5 {
+                completion(.serverError, nil)
+                return
+            }
+            
+            completion(.unknownError, nil)
+        }.resume()
+    }
+    
+    private func buildAccessTokenRequest(for username: String, password: String) -> URLRequest? {
+        guard let clientId = clientId, let serverURL = serverURL else {
+            return nil
+        }
+        
+        var formURLString = "username=\(username)"
+        
+        formURLString = formURLString.appending("&password=\(password)")
```

Comment:
do you need to url encode the username and password since they are user provided?

what happens if a user's password contains `&`